### PR TITLE
add support for gtk

### PR DESCRIPTION
**Pull request by getchoo**
this adds support for the [gtk](https://github.com/catppuccin/gtk) theme, as well as an accent option for future use in other modules :)

### CODE REVIEW

File: modules/home-manager/gtk.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    };
+  };
+
+  config.gtk.theme = with lib;
+    with builtins;
+    let
+      # string -> string
+      # this capitalizes the first letter in a string
+      # it's used to set the theme name correctly here
+      mkUpper = word: (toUpper (substring 0 1 word)) + (substring 1 999 word);
```
Comment (Stonks3141) [inline modules/home-manager/gtk.nix:34]:
```suggestion
      mkUpper = word: (toUpper (substring 0 1 word)) + (substring 1 (stringLength word - 1) word);
```

File: modules/home-manager/gtk.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      description = "Catppuccin accents for gtk";
+    };
+    size = mkOption {
+      type = types.enum [ "standard" "compact" ];
+      default = "standard";
+      description = "Catppuccin size variant for gtk";
+    };
+    tweaks = mkOption {
+      type = types.listOf (types.enum [ "black" "rimless" "normal" ]);
+      default = ["normal"];
```
Comment (Stonks3141) [inline modules/home-manager/gtk.nix:23]:
```suggestion
      default = [ "normal" ];
```

File: modules/home-manager/gtk.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+      # use the light gtk theme for latte
+      gtkTheme = if cfg.flavour == "latte" then "Light" else "Dark";
+
+    in mkIf cfg.enable {
+      name =
+        "Catppuccin-${flavourUpper}-${sizeUpper}-${accentUpper}-${gtkTheme}";
+      package = pkgs.catppuccin-gtk.override {
+        inherit (cfg) size tweaks;
+        accents = [cfg.accent];
```
Comment (Stonks3141) [inline modules/home-manager/gtk.nix:48]:
```suggestion
        accents = [ cfg.accent ];
```

**Review by Stonks3141 — CHANGES_REQUESTED** [review]
LGTM! I don't know if this is just what the formatter you're using does, but there is no whitespace around single-element arrays. If that is a formatter thing, ignore those suggestions and we'll figure that out when we decide on a formatter.

**Comment by getchoo** [comment]
> I don't know if this is just what the formatter you're using does, but there is no whitespace around single-element arrays. If that is a formatter thing, ignore those suggestions and we'll figure that out when we decide on a formatter.

that was just my mistake lol

**Pull request merged**